WeCantSpell.Hunspell/LineReader.cs

```diff
@@ -413,28 +417,28 @@ private void ChangeEncoding(Encoding newEncoding)
             // The characters that were loaded after the encoding change need to be re-decoded
             for (var bufferIndex = _position.BufferIndex; bufferIndex < _buffers.Count; bufferIndex++)
             {
-                var oldCharacters = _buffers[bufferIndex].Memory;
-
-                if (_position.BufferIndex == bufferIndex)
-                {
-                    oldCharacters = oldCharacters.Slice(_position.SubIndex);
-                }
+                var offset = _position.BufferIndex == bufferIndex ? _position.SubIndex : 0;
 
 #if NO_ENCODING_SPANS
-                var restoredBytes = oldEncoding.GetBytes(oldCharacters.ToArray());
+
+                var oldBuffer = _buffers[bufferIndex];
+                var restoredBytes = oldEncoding.GetBytes(oldBuffer.GetRawBuffer(), offset, oldBuffer.Length - offset);
```

Comment:
I think it would be safer to move `GetBytes` into the buffer type when `NO_ENCODING_SPANS` to keep the raw buffer private.

Reply:
I rewrote it to be safer in general